data/dataexchange/src/main/java/tech/pegasys/teku/data/SlashingProtectionImporter.java

```diff
@@ -129,35 +129,56 @@ private SigningHistory signingHistoryConverter(final SigningHistory signingHisto
     }
   }
 
-  public void updateLocalRecords(final Path slashingProtectionPath) {
+  public void updateLocalRecords(
+      final Path slashingProtectionPath, final Consumer<String> statusConsumer) {
     this.slashingProtectionPath = slashingProtectionPath;
-    data.forEach(this::updateLocalRecord);
-    log.display("Updated " + data.size() + " validator slashing protection records");
+    data.forEach(record -> updateLocalRecord(record, statusConsumer));
+    statusConsumer.accept("Updated " + data.size() + " validator slashing protection records");
   }
 
-  private void updateLocalRecord(final SigningHistory signingHistory) {
+  public Optional<String> updateSigningRecord(final BLSPublicKey publicKey) {
+    final BLSPubKey key = new BLSPubKey(publicKey);
+    data.stream()
+        .filter(signingHistory -> signingHistory.pubkey.equals(key))
+        .collect(Collectors.toList())
+        .forEach(record -> updateLocalRecord(record, (__) -> {}));
```

Comment:
Should be able to just iterate the stream rather than having to collect into a list first.

```suggestion
        .forEach(record -> updateLocalRecord(record, __ -> {}));
```